app/models/document.rb

```diff
@@ -251,6 +251,11 @@ def send_email_on_publish?
     update_type == "major"
   end
 
+  def add_attachment(document, attachment)
```

Comment:
I think the document here refers to the class instance itself? So should the method be:
```
def add_attachment(attachment):
  attachments ||= []
  attachments.push(attachment)
end
```